src/shared/clients/iotClient.ts

```diff
@@ -0,0 +1,676 @@
+        getLogger().debug('ListThingCertificates returned response: %O', response)
+        return { certificates: filteredCerts, nextToken: nextToken }
+    }
+
+    /**
+     * Lists Things attached to specified certificate.
+     *
+     * @throws Error if there is an error calling IoT.
+     */
+    public async listThingsForCert(request: Iot.ListPrincipalThingsRequest): Promise<string[]> {
+        getLogger().debug('UpdateCertificate called with request: %O', request)
+        const iot = await this.createIot()
+
+        let iotThings: Iot.ThingName[]
+        try {
+            const output = await iot
+                .listPrincipalThings({
+                    maxResults: request.maxResults ?? DEFAULT_MAX_THINGS,
+                    nextToken: request.nextToken,
+                    principal: request.principal,
+                })
+                .promise()
+            iotThings = output.things ?? []
+        } catch (e) {
+            getLogger().error('Failed to list things: %O', e)
+            throw e
+        }
+
+        getLogger().debug('ListThings returned response: %O', iotThings)
+        return iotThings
+    }
+
+    /**
+     * Creates an X.509 certificate with a 2048 bit RSA keypair and saves them
+     * to the filesystem.
+     *
+     * @throws Error if there is an error calling IoT.
+     */
+    public async createCertificateAndKeys(request: CreateCertificateRequest): Promise<void> {
+        getLogger().debug('CreateCertificate called with request: %O', request)
+        const iot = await this.createIot()
+
+        let certId: string | undefined
+        let certPem: string | undefined
+        let privateKey: string | undefined
+        let publicKey: string | undefined
+        try {
+            const output = await iot
+                .createKeysAndCertificate({
+                    setAsActive: request.active,
+                })
+                .promise()
+            certId = output.certificateId
+            certPem = output.certificatePem
+            privateKey = output.keyPair?.PrivateKey
+            publicKey = output.keyPair?.PublicKey
+        } catch (e) {
+            getLogger().error('Failed to create certificate and keys: %O', e)
+            throw e
+        }
+
+        if (!certPem || !privateKey || !publicKey) {
+            getLogger().error('Could not download certificate')
+            return undefined
+        }
+
+        //Save resources
+        try {
+            await fs.writeFile(request.certPath, certPem, { encoding: PEM_FILE_ENCODING, mode: MODE_RW_R_R })
+            await fs.writeFile(request.privateKeyPath, privateKey, { encoding: PEM_FILE_ENCODING, mode: MODE_RW_R_R })
+            await fs.writeFile(request.publicKeyPath, publicKey, { encoding: PEM_FILE_ENCODING, mode: MODE_RW_R_R })
+        } catch (e) {
+            getLogger().error('Failed to write files: %O', e)
+            throw e
+        }
```

Comment:
I don't like this being wrapped up with the client. This should be handled by the caller.